includes/wizards/class-setup-wizard.php

```diff
@@ -277,14 +277,98 @@ public function api_retrieve_theme_and_set_defaults() {
+		<!-- /wp:spacer -->
+
+		<!-- wp:columns -->
+		<div class="wp-block-columns"><!-- wp:column {"verticalAlignment":"center","width":"25%"} -->
+		<div class="wp-block-column is-vertically-aligned-center" style="flex-basis:25%"><!-- wp:heading -->
+		<h2>' . __( 'Support our publication', 'newspack' ) . '</h2>
+		<!-- /wp:heading --></div>
+		<!-- /wp:column -->
+
+		<!-- wp:column {"width":"75%"} -->
+		<div class="wp-block-column" style="flex-basis:75%"><!-- wp:newspack-blocks/donate /--></div>
+		<!-- /wp:column --></div>
+
+		<!-- wp:spacer {"height":20} -->
+		<div style="height:20px" aria-hidden="true" class="wp-block-spacer"></div>
+		<!-- /wp:spacer --></div></div>
+		<!-- /wp:group -->
+		' : '';
+		$image_url       = 'https://picsum.photos/id/424/600/400	';
+		$subscribe_block = $has_newsletters ? '<!-- wp:group {"className":"newspack-pattern subscribe__style-1"} -->
+		<div class="wp-block-group newspack-pattern subscribe__style-1"><div class="wp-block-group__inner-container"><!-- wp:media-text {"align":"","mediaPosition":"right","mediaId":131,"mediaType":"image","verticalAlignment":"center","imageFill":true,"backgroundColor":"light-gray"} -->
+		<div class="wp-block-media-text has-media-on-the-right is-stacked-on-mobile is-vertically-aligned-center is-image-fill has-light-gray-background-color has-background"><figure class="wp-block-media-text__media" style="background-image:url(' . $image_url . ');background-position:50% 50%"><img src="' . $image_url . '" alt="" class="wp-image-131 size-full"/></figure><div class="wp-block-media-text__content"><!-- wp:spacer {"height":32} -->
+		<div style="height:32px" aria-hidden="true" class="wp-block-spacer"></div>
+		<!-- /wp:spacer -->
+
+		<!-- wp:heading -->
+		<h2>' . __( 'Subscribe', 'newspack' ) . '</h2>
+		<!-- /wp:heading -->
+
+		<!-- wp:jetpack/mailchimp {"interests":[]} -->
+		<!-- wp:jetpack/button {"element":"button","uniqueId":"mailchimp-widget-id","text":"Join my email list"} /-->
+		<!-- /wp:jetpack/mailchimp -->
+
+		<!-- wp:spacer {"height":32} -->
+		<div style="height:32px" aria-hidden="true" class="wp-block-spacer"></div>
+		<!-- /wp:spacer --></div></div>
+		<!-- /wp:media-text --></div></div>
+		<!-- /wp:group -->' : '';
+		return require $file_path;
+	}
+
+	/**
+	 * Get homepage patterns registered by the Newspack Blocks plugin.
+	 *
+	 * @param int $index Pattern ID.
+	 * @return array|object|bool Homepage pattern (false if not found) or all patterns.
+	 */
+	private function get_homepage_patterns( $index = null ) {
+		$pattern_file_base = dirname( NEWSPACK_PLUGIN_FILE ) . '/includes/templates/block-patterns/homepage/';
+		$homepage_patterns = [];
+		if ( null !== $index ) {
+			$file_path = $pattern_file_base . ( $index + 1 ) . '.php';
+			if ( file_exists( $file_path ) ) {
+				return $this->load_homepage_pattern( $file_path );
+			} else {
+				return false;
+			}
+		}
+		$block_patterns_amount = 12;
```

Comment:
This seems fragile to me if we were to add or remove patterns. Can we get this number by counting the number of pattern files in the `block-patterns` directory? Also by possibly reading the files in the directory and building the paths from those filenames instead of reconstructing the filenames using an iterator?

Reply:
Good idea! Added in 719de6697b5c3ae67880c6fed02b7db393a5627a

Reply:
Hmm, after this change, previewing and saving no longer seem to work. I can "activate" different patterns in the onboarding step, but it doesn't actually change the pattern in the preview modal or on the front-end.

Reply:
Whoops, my bad. Fixed in 8928e1680d6296a26f079f23fedd82c840517761

Reply:
That fixed it!